homeassistant/components/velbus/light.py

```diff
@@ -53,25 +73,46 @@ def brightness(self):
 
     def turn_on(self, **kwargs):
         """Instruct the Velbus light to turn on."""
-        try:
-            if ATTR_BRIGHTNESS in kwargs:
-                self._module.set_dimmer_state(
-                    self._channel,
-                    kwargs[ATTR_BRIGHTNESS],
-                    kwargs.get(ATTR_TRANSITION, 0),
-                )
+        if self._module.light_is_buttonled(self._channel):
+            if ATTR_FLASH in kwargs:
+                if kwargs[ATTR_FLASH] == FLASH_LONG:
+                    action = "slow"
+                elif kwargs[ATTR_FLASH] == FLASH_SHORT:
+                    action = "fast"
+                else:
+                    action = "on"
             else:
-                self._module.restore_dimmer_state(
-                    self._channel, kwargs.get(ATTR_TRANSITION, 0),
-                )
-        except VelbusException as err:
-            _LOGGER.error("A Velbus error occurred: %s", err)
+                action = "on"
+            try:
+                self._module.set_led_state(self._channel, action)
+            except VelbusException as err:
+                _LOGGER.error("A Velbus error occurred: %s", err)
+        else:
+            try:
+                if ATTR_BRIGHTNESS in kwargs:
+                    self._module.set_dimmer_state(
+                        self._channel,
+                        kwargs[ATTR_BRIGHTNESS],
+                        kwargs.get(ATTR_TRANSITION, 0),
+                    )
+                else:
+                    self._module.restore_dimmer_state(
+                        self._channel, kwargs.get(ATTR_TRANSITION, 0),
+                    )
+            except VelbusException as err:
+                _LOGGER.error("A Velbus error occurred: %s", err)
 
     def turn_off(self, **kwargs):
         """Instruct the velbus light to turn off."""
-        try:
-            self._module.set_dimmer_state(
-                self._channel, 0, kwargs.get(ATTR_TRANSITION, 0),
-            )
-        except VelbusException as err:
-            _LOGGER.error("A Velbus error occurred: %s", err)
+        if self._module.light_is_buttonled(self._channel):
+            try:
+                self._module.set_led_state(self._channel, "off")
+            except VelbusException as err:
+                _LOGGER.error("A Velbus error occurred: %s", err)
+        else:
+            try:
+                self._module.set_dimmer_state(
```

Comment:
We can save some lines by using `getattr`:
```py
if self._module.light_is_buttonled(self._channel):
    attr, *args = "set_led_state", self._channel, "off"
else:
    attr, *args = "set_dimmer_state", self._channel, 0, kwargs.get(ATTR_TRANSITION, 0)
try:
    getattr(self._module, attr)(*args)
except VelbusException as err:
    _LOGGER.error("A Velbus error occurred: %s", err)
```

Reply:
It actually results in an extra line due to black formatting. Is it still desirable to adjust?

        if self._module.light_is_buttonled(self._channel):
            attr, *args = "set_led_state", self._channel, "off"
        else:
            attr, *args = (
                "set_dimmer_state",
                self._channel,
                0,
                kwargs.get(ATTR_TRANSITION, 0),
            )
        try:
            getattr(self._module, attr)(*args)
        except VelbusException as err:
            _LOGGER.error("A Velbus error occurred: %s", err)

Reply:
Ok. Yeah, I still think it's easier to read since there will be less indentation blocks and more dry code.

Reply:
Fixed in [this ](https://github.com/home-assistant/home-assistant/pull/30445/commits/23b3e5f5c96fd29b03496bc926a03e4a3d167036) commit